### PR TITLE
omit fields where fact is nil

### PR DESCRIPTION
**Pull request by laviro**

### CODE REVIEW

File: lib/foreman_inventory_upload/generators/json_stream.rb [hunk 1]
```diff
@@ -26,8 +26,14 @@ def raw(string)
         @out << string
       end
 
-      def simple_field(name, value, last = false)
-        @out << "\"#{name}\": #{stringify_value(value)}#{last ? '' : ','}" unless value.nil?
+      def simple_field(name, value, last = false, &block)
+        return if value.nil?
+        return if value.kind_of?(String) && value.empty?
+        return if value.kind_of?(Array) && value.compact.empty?
```
Comment (ShimShtein) [inline lib/foreman_inventory_upload/generators/json_stream.rb:32]:
```suggestion
        return if value.empty?
        return if value.kind_of?(Array) && value.compact.empty?
```

`empty?` works on simple values.

**Review by ShimShtein — CHANGES_REQUESTED** [review]

File: lib/foreman_inventory_upload/generators/json_stream.rb (hunk 1 above)
Comment (laviro) [reply lib/foreman_inventory_upload/generators/json_stream.rb:32]: yes but for numbers it can return an error as there is no `empty?` for `Integer`
in case we will want to pass in an Integer (?)

Comment (ShimShtein) [reply lib/foreman_inventory_upload/generators/json_stream.rb:32]: You are right, sorry:
``` ruby
return if value.try(:empty?)
```
will do the trick for types that don't have empty, like Integer.

Comment (laviro) [reply lib/foreman_inventory_upload/generators/json_stream.rb]: Thanks, updated !

**Review by ShimShtein — CHANGES_REQUESTED** [review]
Can we have some tests for these cases?
We can either add direct tests for `json_stream` or we can add more test cases in `slice_gnerator_test` for nil and non-nil values.

**Comment by laviro** [comment]
Added tests :)

**Comment by san7ket** [comment]
No validations issues from Yupana or HBI. HBI sets the fields to "None" for some reason though doesn't omit them. maybe these are default/required fields.

**Comment by ShimShtein** [comment]
Merged, thanks @laviro and @san7ket!
